app/controllers/api/admin/chemicals_controller.rb

```diff
@@ -0,0 +1,238 @@
+module Api
```

Comment:
@mcumcu this controller looks a bit overloaded with multiple responsibilities. I think we can move some validations on model or db level.